File: magellan-library/src/main/java/com/wealthfront/magellan/transitions/NoAnimationTransition.java
package com.wealthfront.magellan.transitions;

import android.view.View;

import com.wealthfront.magellan.Direction;
import com.wealthfront.magellan.NavigationType;

public class NoAnimationTransition implements Transition {

  @Override
  public void animate(
      View from, View to, NavigationType navType, Direction direction, Callback callback) {
    callback.onAnimationEnd();
  }

}

File: magellan-library/src/main/java/com/wealthfront/magellan/transitions/NoAnimationTransition.kt
package com.wealthfront.magellan.transitions

import android.view.View
import com.wealthfront.magellan.Direction

public class NoAnimationTransition : MagellanTransition {

  override fun interrupt() {}

  override fun animate(from: View?, to: View, direction: Direction, onAnimationEndCallback: () -> Unit) {
    onAnimationEndCallback()
  }
}
